core/server/worker/src/test/java/alluxio/worker/page/PagedBlockStoreCommitBlockTest.java

```diff
@@ -0,0 +1,261 @@
+    } catch (IOException e) {
+      throw new RuntimeException();
+    }
+
+    try {
+      prepareBlockStore();
+    } catch (Exception e) {
+      throw e;
+    }
+
+    assertThrows(RuntimeException.class, () -> {
+      mPagedBlockStore.commitBlock(SESSION_ID, BLOCK_ID, false);
+    });
+
+    verify(mListener, never()).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+    verify(mListener, never()).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+
+    try {
+      mPagedBlockStore.close();
+    } catch (Exception e) {
+      throw e;
+    }
+  }
+
+  // This Test case success commitToLocal, expecting one exception,
+  // and only one onCommit method should be called.
+  @Test
+  public void localCommitSuccessAndMasterCommitFail() throws IOException, InterruptedException {
+    try {
+      doAnswer((i) -> {
+        throw new AlluxioStatusException(Status.UNAVAILABLE);
+      }).when(mMockedBlockMasterClient).commitBlock(anyLong(), anyLong(), anyString(),
+               anyString(), anyLong(), anyLong());
+    } catch (AlluxioStatusException e) {
+      throw new RuntimeException();
+    }
+    try {
+      mPageMetaStore = new PagedBlockMetaStore(mDirs) {
+        // here commit always success
+        @Override
+        public PagedBlockMeta commit(long BLOCK_ID) {
+          return super.commit(BLOCK_ID);
+        }
+      };
+      mCacheManager = CacheManager.Factory.create(mConf, mCacheManagerOptions, mPageMetaStore);
+
+      mPagedBlockStore = new PagedBlockStore(mCacheManager, mUfs, mBlockMasterClientPool, mWorkerId,
+               mPageMetaStore, mCacheManagerOptions.getPageSize());
+    } catch (IOException e) {
+      throw new RuntimeException();
+    }
+
+    try {
+      prepareBlockStore();
+    } catch (Exception e) {
+      throw e;
+    }
+
+    assertThrows(RuntimeException.class, () -> {
+      mPagedBlockStore.commitBlock(SESSION_ID, BLOCK_ID, false);
+    });
+    verify(mListener).onCommitBlockToLocal(anyLong(), any(BlockStoreLocation.class));
+    verify(mListener, never()).onCommitBlockToMaster(anyLong(), any(BlockStoreLocation.class));
+  }
+
+  // Prepare PageBlockStore and creat a temp block for following test
+  public void prepareBlockStore() throws IOException, InterruptedException {
+    PagedBlockStoreDir dir =
+            (PagedBlockStoreDir) mPageMetaStore.allocate(BlockPageId.tempFileIdOf(BLOCK_ID), 1);
+
+    dir.putTempFile(BlockPageId.tempFileIdOf(BLOCK_ID));
+    PagedTempBlockMeta blockMeta = new PagedTempBlockMeta(BLOCK_ID, dir);
+    mPagedBlockStore.createBlock(SESSION_ID, BLOCK_ID, OFFSET,
+             new CreateBlockOptions(null, null, mBlockSize));
+    byte[] data = new byte[mBlockSize];
+    Arrays.fill(data, (byte) 1);
+    ByteBuffer buf = ByteBuffer.wrap(data);
+    try (BlockWriter writer = mPagedBlockStore.createBlockWriter(SESSION_ID, BLOCK_ID)) {
+      Thread.sleep(1000);
```

Comment:
Thread.sleep is flaky 
can we use common utils wait for family functions? 
https://github.com/TachyonNexus/enterprise/blob/master/core/common/src/main/java/alluxio/util/CommonUtils.java#L319-L359